app/com/gu/floodgate/reindex/BulkJobActor.scala

```diff
@@ -162,17 +163,18 @@ class BulkJobActor(contentSources: List[ContentSource], runningJobService: Runni
   }
   private def getCompletedJobs: Future[List[CompletedJobInfo]] = {
 
-    val completedSources = orderedContentSources.filterNot(source => pendingReindexes.contains(source))
-    val result = Future.sequence(completedSources.map { source =>
+    val completedAndRunningSources = orderedContentSources.filterNot(source => pendingReindexes.contains(source))
+
+    val result = Future.sequence(completedAndRunningSources.map { source =>
       jobHistoryService.getLatestJob(source.id, source.environment).map { maybeJobHistory =>
-        maybeJobHistory.map { jobHistory =>
-          CompletedJobInfo(source.appName, source.id, source.environment, jobHistory.startTime, jobHistory.finishTime, jobHistory.status)
+        maybeJobHistory.collect {
+          case jh if runningJobService.getRunningJob(jh.contentSourceId, jh.environment).isLeft =>
+            CompletedJobInfo(source.appName, source.id, source.environment, jh.startTime, jh.finishTime, jh.status)
```

Comment:
`CompletedJobInfo` could have an apply method on it's companion object which takes a `ContentSource` and `JobHistory` object in order to create a `CompletedJobInfo`.

Reply:
Good idea!